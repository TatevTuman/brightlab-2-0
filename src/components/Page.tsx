import { PageProps } from 'gatsby'
import React, { memo } from 'react'

interface PageComponentProps extends PageProps {}

const Page: React.FC<PageComponentProps> = props => {
  const { children } = props

  return (
    <main className="flex-1">
      <div className="container">{children}</div>
    </main>
  )
}

export default memo(Page)
