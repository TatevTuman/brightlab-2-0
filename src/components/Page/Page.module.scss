.page {
  padding-top: 116px; // Header height
  min-height: 100vh;
}
