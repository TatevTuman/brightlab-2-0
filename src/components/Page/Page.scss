.page {
  padding-top: 116px; // Header height
}
