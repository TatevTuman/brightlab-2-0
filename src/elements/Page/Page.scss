.page {

}

.page[data-name="Home"] {

  .logo {
    position: absolute;
    top: 0;
    right: 0;
  }
}
