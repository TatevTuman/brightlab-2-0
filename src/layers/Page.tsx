import React, { memo, useState } from 'react'
import { RouteComponentProps } from '@reach/router'
import { ApolloClient } from '@apollo/client'

export interface PageLayerMethods {}
export interface PageLayerState {
  page: boolean
}
export interface PageLayerProps extends RouteComponentProps {
  client: ApolloClient<Record<string, any>>
  children(page: {
    state: PageLayerState
    props: Omit<PageLayerProps, 'children' | 'client'>
    pageMethods: PageLayerMethods
  }): JSX.Element | JSX.Element[]
}

const PageLayer: React.FC<PageLayerProps> = props => {
  const { children, client, ...otherProps } = props
  const [state, setState] = useState<PageLayerState>({ page: true })

  const pageMethods: PageLayerMethods = {}

  return <>{children({ state, props: otherProps, pageMethods })}</>
}

export default memo(PageLayer)
