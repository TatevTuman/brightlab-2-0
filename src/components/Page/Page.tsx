import React, { memo } from 'react'
import { RouteComponentProps } from '@reach/router'
import { Container } from '@components'
import { useSiteMetadata } from '@hooks'
import './Page.scss'

interface PageProps extends RouteComponentProps {
  children: JSX.Element | JSX.Element[]
}

const Page: React.FC<PageProps> = props => {
  const { children, path } = props
  const { navigation } = useSiteMetadata()

  const isDynamicPage = path!.includes('/*')
  const pathWithoutSlash = path!.slice(0, isDynamicPage ? -2 : -1)
  const page = navigation.find(page => page.path === pathWithoutSlash)

  return (
    <div className="page">
      <Container>{children}</Container>
    </div>
  )
}

Page.defaultProps = {
  path: ''
}

export default memo(Page)
