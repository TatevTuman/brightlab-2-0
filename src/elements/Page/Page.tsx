import React from 'react'
import { Container, SEO } from '@components'
import './Page.scss'

interface PageProps {
  name: string
  children: JSX.Element
}

const Page: React.FC<PageProps> = props => {
  const { name, children } = props

  return (
    <section className="page">
      <SEO title={name} />
      <Container>{children}</Container>
    </section>
  )
}

export default Page
