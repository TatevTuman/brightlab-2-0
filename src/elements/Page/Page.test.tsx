import React from 'react'
import renderer from 'react-test-renderer'
import Page from './Page'

describe('Page', () => {
  it('render()', () => {
    const button = renderer
      .create(
        <Page>
          <div>Hello World</div>
        </Page>
      )
      .toJSON()
    expect(button).toMatchSnapshot()
  })
})
