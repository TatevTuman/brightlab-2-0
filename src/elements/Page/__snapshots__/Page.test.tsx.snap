// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page render() 1`] = `
<div
  className="page"
  style={
    Object {
      "paddingTop": "nullpx",
    }
  }
>
  <div
    className="container"
  >
    <div>
      Hello World
    </div>
  </div>
</div>
`;
